crates/router/src/core/authentication/utils.rs

```diff
@@ -221,3 +222,52 @@ where
     .to_payment_failed_response()?;
     Ok(router_data)
 }
+
+pub async fn get_authentication_connector_data(
+    state: &AppState,
+    key_store: &domain::MerchantKeyStore,
+    business_profile: &storage::BusinessProfile,
+) -> RouterResult<(
+    api_models::enums::AuthenticationConnectors,
+    payments::helpers::MerchantConnectorAccountType,
+)> {
+    let authentication_details: api_models::admin::AuthenticationConnectorDetails =
+        business_profile
+            .authentication_connector_details
+            .clone()
+            .get_required_value("authentication_details")
+            .change_context(errors::ApiErrorResponse::UnprocessableEntity {
+                message: "authentication_connector_details is not available in business profile".into(), 
+            })
```

Comment:
do we validate this struct format when the merchant connector account is created? If so then we should be raising an internal server error here